### PR TITLE
NEML Benchmark tests

### PR DESCRIPTION
**Pull request by acasagran**
Adding simple single element test cases for linear strain hardening and power law creep material models for NEML. These tests are based on the tensor mechanics regression tests and are also included for comparison with NEML. Issue #59.

### CODE REVIEW

File: test/tests/neml_benchmark/linear_strain_hardening/gold/elas_plas_LSH_out.csv [hunk 1]
```diff
@@ -0,0 +1,22 @@
+time,_dt,disp_x1,disp_x4,disp_y1,disp_y4,eff_plastic_strain,el_strain_xx,el_strain_yy,el_strain_zz,pl_strain_xx,pl_strain_yy,pl_strain_zz,pressure,stress_xx,stress_xy,stress_yy,stress_zz,tot_strain_xx,tot_strain_yy,tot_strain_zz,vonmises
+0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0
+0.1,0.1,0,0,0,2.08e-06,0,-6.9333261210831e-07,2.0799978368803e-06,0,0,0,0,231110.87079534,5.5538490414619e-05,0,554666.08985327,138666.5224772,-6.9333261210831e-07,2.0799978368803e-06,0,499969.25688941
```
Comment (bwspenc) [inline test/tests/neml_benchmark/linear_strain_hardening/gold/elas_plas_LSH_out.csv]: It looks like these results differ from the native MOOSE ones by many orders of magnitude. We should have the two models use the same units -- I can't think of a reason why not to. These are just made-up materials used for testing the models.

File: test/tests/neml_benchmark/linear_strain_hardening/plot_stress_strain.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+plt.rcParams.update({'font.size':10})
+data1 = np.genfromtxt('elas_plas_LSH_out.csv', delimiter=',', names=True)
+data2 = np.genfromtxt('elas_plas_LSH_neml_out.csv', delimiter=',', names=True)
+
+fig = plt.figure()
+ax = fig.add_subplot(111)
+ax.grid(color='dimgray', linestyle='--', linewidth=1, alpha=0.5)
+plt.ticklabel_format(style='plain', axis='y', scilimits=(0,0))
+ax.plot(data1['tot_strain_yy']*100.0, data1['vonmises']/1e6, linestyle='-', marker='x', color='red', linewidth=1, label='Bison')
```
Comment (bwspenc) [inline test/tests/neml_benchmark/linear_strain_hardening/plot_stress_strain.py]: Bison -> MOOSE (or BlackBear, whatever you prefer)

File: test/tests/neml_benchmark/power_law_creep/plot_strain_time.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+plt.rcParams.update({'font.size':10})
+data1 = np.genfromtxt('nafems_test5a_lim_out.csv', delimiter=',', names=True)
+data2 = np.genfromtxt('nafems_test5a_lim_neml_out.csv', delimiter=',', names=True)
+
+fig = plt.figure()
+ax = fig.add_subplot(111)
+ax.grid(color='dimgray', linestyle='--', linewidth=1, alpha=0.5)
+plt.ticklabel_format(style='plain', axis='y', scilimits=(0,0))
+ax.plot(data1['time'], data1['eps_crp_xx'], linestyle='-', marker='s', markersize=3, color='blue', linewidth=1, label='$\epsilon_{xx}$ (Bison)')
```
Comment (bwspenc) [inline test/tests/neml_benchmark/power_law_creep/plot_strain_time.py]: Bison->MOOSE

File: test/tests/neml_benchmark/power_law_creep/power_law_creep.xml [hunk 4]
```diff
@@ -0,0 +1,42 @@
+<!-- Units are MPa, mm, s, and Kelvin -->
+<materials>
+  <!-- Nominally valid over 273 to 1073 K -->
```
Comment (bwspenc) [inline test/tests/neml_benchmark/power_law_creep/power_law_creep.xml]: Remove irrelevant comments

File: test/tests/neml_benchmark/power_law_creep/nafems_test5a_lim_neml.i [hunk 5]
```diff
@@ -0,0 +1,334 @@
...
+  petsc_options_value = ' lu       superlu_dist'
+
+  line_search = 'none'
+
+  nl_rel_tol = 1e-5
+  nl_abs_tol = 1e-8
+  l_max_its = 50
+  nl_max_its = 100
+  end_time = 1000.0
+  num_steps = 10000
```
Comment (bwspenc) [inline test/tests/neml_benchmark/power_law_creep/nafems_test5a_lim_neml.i]: Get rid of num_steps

File: test/tests/neml_benchmark/linear_strain_hardening/elas_plas_LSH.i [hunk 6]
```diff
@@ -0,0 +1,232 @@
...
+[AuxVariables]
+  [./eff_plastic_strain]
+    order = CONSTANT
+    family = MONOMIAL
+  [../]
+[]
+
+[Modules/TensorMechanics/Master]
+  [./all]
+    strain = FINITE
```
Comment (bwspenc) [inline test/tests/neml_benchmark/linear_strain_hardening/elas_plas_LSH.i]: See if setting this to SMALL still matches

File: test/tests/neml_benchmark/power_law_creep/nafems_test5a_lim.i [hunk 7]
```diff
@@ -0,0 +1,238 @@
...
+[AuxVariables]
+  [./creep]
+    order = CONSTANT
+    family = MONOMIAL
+  [../]
+[]
+
+[Modules/TensorMechanics/Master]
+  [./all]
+    strain = FINITE
```
Comment (bwspenc) [inline test/tests/neml_benchmark/power_law_creep/nafems_test5a_lim.i]: Try SMALL

File: test/tests/neml_benchmark/power_law_creep/nafems_test5a_lim_neml.i [hunk 8]
```diff
@@ -0,0 +1,333 @@
...
+  [../]
+  [./elastic_strain_zz]
+    order = CONSTANT
+    family = MONOMIAL
+  [../]
+[]
+
+[Kernels]
+  [./TensorMechanics]
+    use_displaced_mesh = true
```
Comment (bwspenc) [inline test/tests/neml_benchmark/power_law_creep/nafems_test5a_lim_neml.i]: this should be false

File: test/tests/neml_benchmark/power_law_creep/nafems_test5a_lim_neml.i [hunk 9]
```diff
@@ -0,0 +1,333 @@
...
+    variable = disp_x
+    boundary = left
+    value = 0.0
+  [../]
+  [./top_press]
+    type = Pressure
+    variable = disp_y
+    boundary = top
+    component = 1
+    factor = -100.0
```
Comment (bwspenc) [inline test/tests/neml_benchmark/power_law_creep/nafems_test5a_lim_neml.i]: You need to set `use_displaced_mesh=false` in these.

File: test/tests/neml_benchmark/linear_strain_hardening/elas_plas_LSH.i (hunk 6 above)
Comment (acasagran) [reply test/tests/neml_benchmark/linear_strain_hardening/elas_plas_LSH.i]: Tried this and it didn't work. Do I need to make other changes as well?

**Comment by bwspenc** [comment]
@acasagran I just pushed a commit to your branch that makes the NEML and native model more consistent with each other. They all use finite strain now, and have the same tolerances. I set the NEML creep model to only run for 30 steps because it starts failing shortly after that.  We'll need further work on that, but I think it's good enough for now. I had to re-gold a few of the CSV files because of the changes. If you could review my changes and are good with them, then I'll merge this.

**Comment by acasagran** [comment]
@bwspenc The changes look ok to me. I'd go ahead and merge them.

**Pull request merged**